src/subclassers.ts

```diff
@@ -34,6 +42,46 @@ type FunctionTaking<TIn> = (param: TIn) => any;
 
 type HasTypedFunctionOn<T, K extends string> = Record<K, FunctionTaking<T>>;
 
+export function WithInitFunctional(
```

Comment:
about WithInit stuff
i think it can be like this because i still going to use it as a parameter for main HTPipe like we do this now
so main HTPipe will return the function what accept WithInit as a parameter and it return the main object what builds from all sub-HTPipes